### PR TITLE
Added controller method to retrieve list of commands to publish device telemetry

### PR DESCRIPTION
**Pull request by dashevchenko**
## Pull Request description

Added method  to DeviceController that retrieves list of possible commands to publish device telemetry.

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

**Review by YevhenBondarenko — APPROVED** [review]

File: application/src/main/resources/thingsboard.yml [hunk 1]
```diff
@@ -978,6 +978,22 @@ transport:
     enabled: "${TB_TRANSPORT_STATS_ENABLED:true}"
     print-interval-ms: "${TB_TRANSPORT_STATS_PRINT_INTERVAL_MS:60000}"
 
+# Device connectivity properties to publish telemetry
```
Comment (YevhenBondarenko) [inline application/src/main/resources/thingsboard.yml:981]: Plz add ability to enable/disable per each protocol
security protocols by default should be disabled

File: dao/src/main/java/org/thingsboard/server/dao/device/DeviceConnectivityInfo.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.dao.device;
+
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
```
Comment (YevhenBondarenko) [inline dao/src/main/java/org/thingsboard/server/dao/device/DeviceConnectivityInfo.java:20]: extra line

File: dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java [hunk 3]
```diff
@@ -681,4 +743,115 @@ public EntityType getEntityType() {
...
+                        command.append(" -u ").append(credentials.getUserName());
+                    }
+                    if (credentials.getPassword() != null) {
+                        command.append(" -P ").append(credentials.getPassword());
+                    }
+                } else {
+                    return null;
+                }
+                break;
+            case X509_CERTIFICATE:
```
Comment (YevhenBondarenko) [inline dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java:825]: plz remove x509

File: application/src/main/java/org/thingsboard/server/controller/DeviceController.java [hunk 4]
```diff
@@ -155,6 +161,24 @@ public DeviceInfo getDeviceInfoById(@ApiParam(value = DEVICE_ID_PARAM_DESCRIPTIO
         return checkDeviceInfoId(deviceId, Operation.READ);
     }
 
+    @ApiOperation(value = "Get commands to publish device telemetry (getDevicePublishTelemetryCommands)",
```
Comment (YevhenBondarenko) [inline application/src/main/java/org/thingsboard/server/controller/DeviceController.java:164]: plz add response example for swagger

File: dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java [hunk 5]
```diff
@@ -681,4 +743,115 @@ public EntityType getEntityType() {
         return EntityType.DEVICE;
     }
 
+    private String getHttpPublishCommand(String baseurl, DeviceCredentials deviceCredentials) {
+        if (deviceCredentials.getCredentialsType() == DeviceCredentialsType.ACCESS_TOKEN) {
+            return String.format("curl -v -X POST %s/api/v1/%s/telemetry --header Content-Type:application/json --data " + PAYLOAD,
```
Comment (YevhenBondarenko) [inline dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java:748]: you should use base url only if host from config is null

File: dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java [hunk 6]
```diff
@@ -681,4 +743,115 @@ public EntityType getEntityType() {
...
+
+    private String getMqttPublishCommand(DeviceCredentials deviceCredentials) {
+        return getMqttPublishCommand("v1/devices/me/telemetry", deviceCredentials, " -m " + PAYLOAD);
+    }
+
+    private String getMqttPublishCommand(String deviceTelemetryTopic, DeviceCredentials deviceCredentials, String payload) {
+        DeviceConnectivityInfo mqttProps = deviceConnectivityConfiguration.getConnectivity().get("mqtt");
+
+        StringBuilder command = new StringBuilder("mosquitto_pub -d -q 1");
+        command.append(" -h ").append(mqttProps.getHost());
```
Comment (YevhenBondarenko) [inline dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java:762]: if host is null you should use base url, the same for port. Plz check other protocols!

File: application/src/main/resources/thingsboard.yml [hunk 7]
```diff
@@ -978,6 +978,34 @@ transport:
...
+      host: "${DEVICE_CONNECTIVITY_MQTTS_HOST:localhost}"
+      port: "${DEVICE_CONNECTIVITY_MQTTS_PORT:8883}"
+    coap:
+      enabled: "${DEVICE_CONNECTIVITY_MQTT_ENABLED:true}"
+      host: "${DEVICE_CONNECTIVITY_COAP_HOST:localhost}"
+      port: "${DEVICE_CONNECTIVITY_COAP_PORT:5683}"
+    coaps:
+      enabled: "${DEVICE_CONNECTIVITY_MQTT_ENABLED:false}"
+      host: "${DEVICE_CONNECTIVITY_COAPS_HOST:localhost}"
+      port: "${DEVICE_CONNECTIVITY_COAPS_PORT:5684}"
```
Comment (ViacheslavKlimov) [inline application/src/main/resources/thingsboard.yml:1007]: Double-check the env variables names, you've used DEVICE_CONNECTIVITY_MQTT_ENABLED for mqtts, coap and coaps

File: dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java [hunk 8]
```diff
@@ -115,13 +130,57 @@ public class DeviceServiceImpl extends AbstractCachedEntityService<DeviceCacheKe
...
+
     @Override
     public DeviceInfo findDeviceInfoById(TenantId tenantId, DeviceId deviceId) {
         log.trace("Executing findDeviceInfoById [{}]", deviceId);
         validateId(deviceId, INCORRECT_DEVICE_ID + deviceId);
         return deviceDao.findDeviceInfoById(tenantId, deviceId.getId());
     }
 
+    @Override
+    public Map<String, String> findDevicePublishTelemetryCommands(String baseUrl, Device device) throws URISyntaxException {
```
Comment (ViacheslavKlimov) [inline dao/src/main/java/org/thingsboard/server/dao/device/DeviceServiceImpl.java:144]: Consider to move this functionality to a separate service. DeviceServiceImpl is already pretty big

**Review by ViacheslavKlimov — CHANGES_REQUESTED** [review]

**Review by YevhenBondarenko — APPROVED** [review]

**Pull request merged**